src/realm/object-store/sync/sync_session.cpp

```diff
@@ -78,9 +78,8 @@ constexpr const char SyncError::c_recovery_file_path_key[];
 ///    * WAITING_FOR_ACCESS_TOKEN: if the session tried to enter ACTIVE,
 ///                                but the token is invalid or expired.
 
-void SyncSession::become_active(std::unique_lock<std::mutex>& lock)
+void SyncSession::become_active()
```

Comment:
Several of these functions took a lock parameter just as a way to signal that the mutex needs to be held, which is done better by the annotations.